google-cloud-language/lib/google/cloud/language/service.rb

```diff
@@ -134,6 +118,23 @@ def default_options
           Google::Gax::CallOptions.new kwargs: default_headers
         end
 
+        def default_encoding
+          # memoize the result so the check is only made once
```

Comment:
IIUC (although rare and probably terrible), Users can reassign `Encoding.default_internal`. Memoizing results will cause encodings to fail if on the (very) low chance of someone changing the default encoding directly. That being said, do you think the performance improvements warrant memoizing the results?